### DISCUSSION TITLE
Consider supporting Rocky Linux

### DISCUSSION THREAD

**Discussion by ghost** [googleapis/google-cloud-cpp#6742] [Ideas]
Note: I have no affiliation with Rocky Linux but my friend is on the engineering team.

CentOS 8 will be EOL December 31 2021 and the introduction of Stream has caused an exodus and plethora of CentOS clones to pop up. I suspect in the future we will reach a point where users will start demanding support for their favorite CentOS-clone platform. I think it would be an interesting idea to at some point switch from CentOS to Rocky Linux (or some alternative).

Why not:

- Oracle Linux: this is pretty self-explanatory. It's Oracle.
- Alma Linux: I did not spend much time looking at their product, but their boot image requires disabling selinux and that doesn't inspire much confidence.

Rocky Linux is not production ready yet, but there is a release candidate so it will be ready soon. In theory, since it's supposed to be binary compatible with RHEL, there should be little to no changes required for this to work.

**Comment by coryan** [comment]
I think this makes total sense. With CentOS:8 becoming a streaming distro the CI builds may break at any time because the distro changes. The objective is to support RHEL:8 customers, and customers using "adjacent" distributions.  It seems that Rocky Linux may fit that bill better than CentOS:8.

**Reply by coryan** [reply]
FWIW, we build with Rockylinux as part of our CI.
